test/vmtests/vmtests/test_min_change.py

```diff
@@ -0,0 +1,365 @@
+    logging.debug(f"- machine_type            = {machine_type}")
+    logging.debug(f"- uefi_firmware_binary    = {uefi_firmware_binary}")
+    logging.debug(f"- boot_type               = {boot_type}")
+
+    username = getuser()
+
+    run_image_customizer(
+        docker_client,
+        image_customizer_container_url,
+        input_image,
+        config_path,
+        username,
+        ssh_public_key,
+        output_format,
+        output_image_path,
+        close_list,
+    )
+
+    vm_image = output_image_path
+    if output_format != "iso":
+        diff_image_path = test_temp_dir.joinpath("image-diff.qcow2")
+
+        # Create a differencing disk for the VM.
+        # This will make it easier to manually debug what is in the image itself and what was set during first boot.
+        local_client.run(
+            ["qemu-img", "create", "-F", "qcow2", "-f", "qcow2", "-b", str(output_image_path), str(diff_image_path)],
+        ).check_exit_code()
+
+        # Ensure VM can write to the disk file.
+        os.chmod(diff_image_path, 0o666)
+
+        vm_image = diff_image_path
+
+    # Create VM.
+    vm_name = test_instance_name
+
+    domain_xml = create_libvirt_domain_xml(VmSpec(vm_name, 4096, 4, vm_image), host_os, boot_type, uefi_firmware_binary)
+
+    vm = LibvirtVm(vm_name, domain_xml, libvirt_conn)
+    close_list.append(vm)
+
+    # Start VM.
+    vm.start()
+
+    # Wait for VM to boot by waiting for it to request an IP address from the DHCP server.
+    vm_ip_address = vm.get_vm_ip_address(timeout=30)
+
+    # Connect to VM using SSH.
+    ssh_known_hosts_path = test_temp_dir.joinpath("known_hosts")
+    open(ssh_known_hosts_path, "w").close()
+
+    with SshClient(vm_ip_address, key_path=ssh_private_key_path, known_hosts_path=ssh_known_hosts_path) as vm_ssh:
+        vm_ssh.run("cat /proc/cmdline").check_exit_code()
+
+        os_release_path = test_temp_dir.joinpath("os-release")
+        vm_ssh.get_file(Path("/etc/os-release"), os_release_path)
+
+        with open(os_release_path, "r") as os_release_fd:
+            os_release_text = os_release_fd.read()
+
+            if input_image_azl_release == 2:
+                assert ("ID=mariner" in os_release_text)
+                assert ('VERSION_ID="2.0"' in os_release_text)
+            elif input_image_azl_release == 3:
+                assert ("ID=azurelinux" in os_release_text)
+                assert ('VERSION_ID="3.0"' in os_release_text)
+            else:
+                assert False, "Unexpected image identity in /etc/os-release"
+
+
+def test_min_change_efi_azl2_qcow_output(
+    docker_client: DockerClient,
+    image_customizer_container_url: str,
+    core_efi_azl2: Path,
+    ssh_key: Tuple[str, Path],
+    test_temp_dir: Path,
+    test_instance_name: str,
+    libvirt_conn: libvirt.virConnect,
+    close_list: List[Closeable],
+) -> None:
+    azl_release = 2
+    config_path = TEST_CONFIGS_DIR.joinpath("os-vm-config.yaml")
```

Comment:
It would probably be bad if the image can't boot without needing changes. So, please use `"nochange-config.yaml"` here. This should work since `cloud-init` is default enabled and will auto-configure the network.

Also, AZL2 has a default network configuration (outside of what cloud-init provides). This was removed in AZL3.